### PR TITLE
Reduce Transient Allocations during Bulk Text Output

### PR DESCRIPTION
**Pull request by miniksa**
Make a few changes to memory usage throughout the application to reduce transient allocations from the `big.txt` test from ~213,000 to ~53,000.

## PR Checklist
* [x] Supports #3075
* [x] I work here.
* [x] Tested manually and WPR'd. Test suite should still pass.
* [x] Am core contributor

## Detailed Description of the Pull Request / Additional comments

Transient allocations are those that are new'd, used, then delete'd. Going back and forth to the system allocator for things we're just going to throw away or use rapidly again is a performance detriment. Not only is it a bunch of time to go ask the system with a syscall, it also hits a whole bunch of locks on the allocators. This PR identifies a few places where we were accidentally allocating and didn't mean to or were allocating and freeing just to turn around and allocate again. I chose other strategies to avoid this.

## Validation Steps Performed
- Ran `big.txt` sample (~6MB file) before and after. Observed heap allocations with WPR.

### CODE REVIEW

**Comment by miniksa** [comment]
Before: 
image

After:
image

(Note that the `ConsoleIoThread` is pretty much gone from Transient allocations. The Render thread has bulked up in how many it has done... probably just a matter of circumstance changing the timing behavior between it and the IO thread.)

File: src/server/ApiMessage.h [hunk 1]
```diff
@@ -73,4 +81,6 @@ typedef struct _CONSOLE_API_MSG
     void SetReplyStatus(const NTSTATUS Status);
     void SetReplyInformation(const ULONG_PTR pInformation);
 
+    // DO NOT PUT MORE FIELDS DOWN HERE. They will be stomped by driver fills.
```
Comment (DHowett) [inline src/server/ApiMessage.h:84]: "stomped by driver fills" is as readable as "cant use, below user32"

Comment (DHowett) [reply src/server/ApiMessage.h:84]: propose "The tail end of this structure has a console driver packet copied over it."

Comment (miniksa) [reply src/server/ApiMessage.h:84]: good call, thx

**Review by DHowett — APPROVED** [review]

File: src/buffer/out/AttrRow.cpp [hunk 2]
```diff
@@ -185,14 +185,14 @@ size_t ATTR_ROW::FindAttrIndex(const size_t index, size_t* const pApplies) const
 // - Finds the hyperlink IDs present in this row and returns them
 // Return value:
 // - An unordered set containing the hyperlink IDs present in this row
```
Comment (carlos-zamora) [inline src/buffer/out/AttrRow.cpp:187]:
```suggestion
// - The hyperlink IDs present in this row
```

**Review by carlos-zamora — APPROVED** [review]

File: src/buffer/out/AttrRow.cpp (hunk 2 above)
Comment (miniksa) [reply src/buffer/out/AttrRow.cpp]: Good call, thx.

**Pull request merged**